src/dmd/safe.d

```diff
@@ -198,3 +200,29 @@ bool isSafeCast(Expression e, Type tfrom, Type tto)
     return false;
 }
 
+/*************************************************
+ * Check for unsafe use of `.ptr` or `.funcptr`
+ * Params:
+ *      sc = context
+ *      e = expression for error messages
+ *      id = `ptr` or `funcptr`
+ *      flag = DotExpFlag
+ * Returns:
+ *      true if error
+ */
+bool checkUnsafeDotExp(Scope* sc, Expression e, Identifier id, int flag)
```

Comment:
If you change the code above like so:
```diff
- Expression dotExp(Type mt, Scope* sc, Expression e, Identifier ident, int flag)
+ Expression dotExp(Type mt, Scope* sc, Expression e, Identifier ident, DotExpFlag flag)
```
You should be able to also fix this signature:

```suggestion
bool checkUnsafeDotExp(Scope* sc, Expression e, Identifier id, DotExpFlag flag)
```

Can you do this in this PR, or you prefer to leave it for a future one?

Reply:
That's the way I originally wrote the code, but it resulted in compile errors because of integral promotions.